<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: run ConsensusClusterPlus</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for ConsensusClusterPlus {ConsensusClusterPlus}"><tr><td>ConsensusClusterPlus {ConsensusClusterPlus}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2> run ConsensusClusterPlus</h2>

<h3>Description</h3>

<p>ConsensusClusterPlus function for determing cluster number and class membership by stability evidence.
calcICL function for calculating cluster-consensus and item-consensus.
</p>


<h3>Usage</h3>

<pre>
ConsensusClusterPlus(
d=NULL, maxK = 3, reps=10, pItem=0.8, pFeature=1, clusterAlg="hc",title="untitled_consensus_cluster",
innerLinkage="average", finalLinkage="average", distance="pearson", ml=NULL,
tmyPal=NULL,seed=NULL,plot=NULL,writeTable=FALSE,weightsItem=NULL,weightsFeature=NULL,verbose=F,corUse="everything")

calcICL(res,title="untitled_consensus_cluster",plot=NULL,writeTable=FALSE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>d</code></td>
<td>
<p>data to be clustered; either a data matrix where columns=items/samples and rows are features.  For example, a gene expression matrix of genes in rows and microarrays in columns, or ExpressionSet object, or a distance object (only for cases of no feature resampling) </p>
</td></tr>
<tr valign="top"><td><code>maxK</code></td>
<td>
<p>integer value. maximum cluster number to evaluate.  </p>
</td></tr>
<tr valign="top"><td><code>reps</code></td>
<td>
<p>integer value. number of subsamples.  </p>
</td></tr>
<tr valign="top"><td><code>pItem</code></td>
<td>
<p>numerical value. proportion of items to sample.  </p>
</td></tr>
<tr valign="top"><td><code>pFeature</code></td>
<td>
<p>numerical value. proportion of features to sample.  </p>
</td></tr>
<tr valign="top"><td><code>clusterAlg</code></td>
<td>
<p>character value. cluster algorithm.  'hc' heirarchical (hclust), 'pam' for paritioning around medoids, 'km' for k-means upon data matrix, 'kmdist' for k-means upon distance matrices (former km option), or a function that returns a clustering.  See example and vignette for more details. </p>
</td></tr>
<tr valign="top"><td><code>title</code></td>
<td>
<p> character value for output directory. Directory is created only if plot is not NULL or writeTable is TRUE. This title can be an abosulte or relative path.  </p>
</td></tr>
<tr valign="top"><td><code>innerLinkage</code></td>
<td>
<p>heirarchical linkage method for subsampling. </p>
</td></tr>
<tr valign="top"><td><code>finalLinkage</code></td>
<td>
<p>heirarchical linkage method for consensus matrix. </p>
</td></tr>
<tr valign="top"><td><code>distance</code></td>
<td>
<p>character value. 'pearson': (1 - Pearson correlation), 'spearman' (1 - Spearman correlation), 'euclidean', 'binary', 'maximum', 'canberra', 'minkowski&quot; or custom distance function.   </p>
</td></tr>
<tr valign="top"><td><code>ml</code></td>
<td>
<p>optional. prior result, if supplied then only do graphics and tables.</p>
</td></tr>
<tr valign="top"><td><code>tmyPal</code></td>
<td>
<p>optional character vector of colors for consensus matrix</p>
</td></tr>
<tr valign="top"><td><code>seed</code></td>
<td>
<p>optional numerical value.  sets random seed for reproducible results.</p>
</td></tr>
<tr valign="top"><td><code>plot</code></td>
<td>
<p>character value. NULL - print to screen, 'pdf', 'png', 'pngBMP' for bitmap png, helpful for large datasets.</p>
</td></tr>
<tr valign="top"><td><code>writeTable</code></td>
<td>
<p>logical value. TRUE - write ouput and log to csv.</p>
</td></tr>
<tr valign="top"><td><code>weightsItem</code></td>
<td>
<p>optional numerical vector. weights to be used for sampling items.</p>
</td></tr>
<tr valign="top"><td><code>weightsFeature</code></td>
<td>
<p>optional numerical vector. weights to be used for sampling features.</p>
</td></tr>
<tr valign="top"><td><code>res</code></td>
<td>
<p> result of consensusClusterPlus.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p> boolean. If TRUE, print messages to the screen to indicate progress.  This is useful for large datasets.</p>
</td></tr>
<tr valign="top"><td><code>corUse</code></td>
<td>
<p>optional character value.  specifies how to handle missing data in correlation distances 'everything','pairwise.complete.obs', 'complete.obs' see cor() for description.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ConsensusClusterPlus implements the Consensus Clustering algorithm of Monti, et al (2003) and extends this method with new functionality and visualizations.
Its utility is to provide quantitative stability evidence for determing a cluster count and cluster membership in an unsupervised analysis.
</p>
<p>ConsensusClusterPlus takes a numerical data matrix of items as columns and rows as features.  This function subsamples this matrix according to pItem, pFeature, weightsItem, and weightsFeature, and clusters the data into 2 to maxK clusters by clusterArg clusteringAlgorithm.  Agglomerative heirarchical (hclust) and kmeans clustering are supported by an option see above.  For users wishing to use a different clustering algorithm for which many are available in R, one can supply their own clustering algorithm as a simple programming hook - see the second commented-out example that uses divisive heirarchical clustering.
</p>
<p>For a detailed description of usage, output and images, see the vignette by: openVignette().
</p>


<h3>Value</h3>

<p>ConsensusClusterPlus returns a list of length maxK.  Each element is a list containing consensusMatrix (numerical matrix), consensusTree (hclust), consensusClass (consensus class asssignments).  ConsensusClusterPlus also produces images.
</p>
<p>calcICL returns a list of two elements clusterConsensus and itemConsensus corresponding to cluster-consensus and item-consensus.  See Monti, et al (2003) for formulas.
</p>


<h3>Author(s)</h3>

<p> Matt Wilkerson mdwilkerson@outlook.com
Peter Waltman waltman@soe.ucsc.edu
</p>


<h3>References</h3>

<p>Please cite the ConsensusClusterPlus publication, below, if you use ConsensusClusterPlus in a publication or presentation:
Wilkerson, M.D., Hayes, D.N. (2010). ConsensusClusterPlus: a class discovery tool with confidence assessments and item tracking. Bioinformatics, 2010 Jun 15;26(12):1572-3. 
</p>
<p>Original description of the Consensus Clustering method:
Monti, S., Tamayo, P., Mesirov, J., Golub, T. (2003) Consensus Clustering:
A Resampling-Based Method for Class Discovery and Visualization of Gene
Expression Microarray Data. Machine Learning, 52, 91-118.
</p>


<h3>Examples</h3>

<pre>

# obtain gene expression data
library(Biobase)
data(geneData)
d=geneData
#median center genes
dc = sweep(d,1, apply(d,1,median))

# run consensus cluster, with standard options
rcc = ConsensusClusterPlus(dc,maxK=4,reps=100,pItem=0.8,pFeature=1,title="example",distance="pearson",clusterAlg="hc")

# same as above but with pre-computed distance matrix, useful for large datasets (&gt;1,000's of items)
dt = as.dist(1-cor(dc,method="pearson"))
rcc2 = ConsensusClusterPlus(dt,maxK=4,reps=100,pItem=0.8,pFeature=1,title="example2",distance="pearson",clusterAlg="hc")

# k-means clustering
rcc3 = ConsensusClusterPlus(d,maxK=4,reps=100,pItem=0.8,pFeature=1,title="example3",distance="euclidean",clusterAlg="km")


### partition around medoids clustering with manhattan distance
rcc4 = ConsensusClusterPlus(d,maxK=4,reps=100,pItem=0.8,pFeature=1,title="example3",distance="manhattan",clusterAlg="pam")

## example of custom distance function as hook:
myDistFunc = function(x){ dist(x,method="manhattan")}
rcc5 = ConsensusClusterPlus(d,maxK=4,reps=100,pItem=0.8,pFeature=1,title="example3",distance="myDistFunc",clusterAlg="pam")


##example of clusterAlg as hook:
#library(cluster)
#dianaHook = function(this_dist,k){
#  tmp = diana(this_dist,diss=TRUE)
#  assignment = cutree(tmp,k)
#  return(assignment)  
#}
#rcc6 = ConsensusClusterPlus(d,maxK=6,reps=25,pItem=0.8,pFeature=1,title="example",clusterAlg="dianaHook")




## ICL
resICL = calcICL(rcc,title="example")


</pre>

<hr /><div style="text-align: center;">[Package <em>ConsensusClusterPlus</em> version 1.46.0 <a href="00Index.html">Index</a>]</div>
</body></html>
